pkg/planner/core/expression_rewriter.go

```diff
@@ -332,6 +339,9 @@ type exprRewriterPlanCtx struct {
 	insertPlan *Insert
 
 	rollExpand *logicalop.LogicalExpand
+
+	// columnVisited records the visited columns, which is used for checking columns privilege
+	columnVisited []*ast.ColumnName
```

Comment:
How about optimizing the variable name columnVisited to imply precollection? The current name suggests that columns have already been processed/checked, which might cause confusion.